src/root/Project/Glance/Arrivals.js

```diff
@@ -0,0 +1,229 @@
+        div('.iconcontainer', [
+          span('.spinner.icon-spinner9', {
+            style: {
+              display: !profile.arrival && enterPressed ? 'block' : 'none',
+            },
+          })]),
+        div('.iconcontainer', [
+          span('.icon-check', {
+            style: {
+              display: profile.arrival ? 'block' : 'none',
+            },
+          })]),
+      ]),
+      div('.col-xs-8', [
+        div('.row', [div(profile.email)]),
+        div('', shiftItems),
+        div('.row', [div(
+          profile.arrival ?
+            `Arrived at ${formatTime(profile.arrival.arrivedAt)}` :
+            'Press return to mark as arrived'
+        )]),
+      ])]
+  ).shareReplay(1)
+
+  const card = TitledCard({
+    ...sources,
+    content$,
+    title$: profile$.pluck('fullName'),
+    cardComponent: LargeCard,
+  })
+
+  const queue$ = $.combineLatest(
+    sources.projectKey$, profile$,
+    (projectKey, profile) =>
+      Arrivals.action.create({
+        projectKey,
+        profileKey: profile.$key,
+      }))
+    .sample(enterPressed$.filter(Boolean))
+
+  return {
+    DOM: card.DOM,
+    queue$,
+  }
+}
+
+const ProfileListItem = sources => {
+  const profile$ = sources.item$
+
+  const icon$ = profile$.map(profile =>
+    div('.arrival-portrait', {class: {arrived: profile.arrival}}, [
+      iconSrc(profile.portraitUrl),
+      span('.icon-check.small'),
+    ]))
+
+  const li = isolate(ListItemClickable)({
+    ...sources,
+    title$: profile$.pluck('fullName'),
+    subtitle$: profile$.pluck('email'),
+    leftDOM$: icon$,
+  })
+
+  return {
+    ...li,
+  }
+}
+
+const ProfileList = sources => ({
+  ...List({
+    ...sources,
+    Control$: $.just(ProfileListItem),
+    rows$: sources.profiles$,
+  }),
+  queue$: $.empty(),
+})
+
+const SearchResults = sources => {
+  const oneProfile$ = sources.profiles$
+    .map(propEq('length', 1))
+    .distinctUntilChanged()
+    .shareReplay(1)
+
+  const control$ = oneProfile$.map(oneProfile =>
+    oneProfile ?
+      ProfileView({
+        ...sources,
+        profile$: sources.profiles$.map(head),
+      }) :
+      ProfileList(sources))
+
+  return {
+    DOM: control$.pluck('DOM').switch(),
+    queue$: control$.pluck('queue$').switch(),
+  }
+}
+
+const profileMatchesTerm = (term) =>
+  compose(any(text => text.includes(term)), prop('search'))
+
+const prepareText = compose(
+  split(' '),
+  toLower,
+  join(' ')
+)
+
+const prepareProfile = profile =>
+  ({...profile, search: prepareText(props(['fullName', 'email'], profile))})
+
+const notEmptyStringΩ = complement(anyPass([isEmpty, isNil]))
+
+const ArrivalsTab = unfetchedSources => {
+  const sources = {...unfetchedSources, ...AssignmentsFetcher(unfetchedSources)}
+
+  const preparedProfiles$ = sources.profilesWithArrival$
+    .map(map(prepareProfile))
+    .shareReplay(1)
+
+  const searchBox = SearchBox(sources)
+  const term$ = searchBox.term$
+    .map(unless(Boolean, always('')))
```

Comment:
Currently the search doesn't work if you type in a capital letter for there name. Maybe a simple `.map(t => t.toLowerCase())` would be enough for now?